runtime/src/loader_utils.rs

```diff
@@ -65,63 +65,6 @@ pub fn create_program(bank: &Bank, loader_id: &Pubkey, name: &str) -> Pubkey {
     program_id
 }
 
-pub fn load_and_finalize_program<T: Client>(
-    bank_client: &T,
-    loader_id: &Pubkey,
-    program_keypair: Option<Keypair>,
-    payer_keypair: &Keypair,
-    name: &str,
-) -> (Keypair, Instruction) {
-    let program = load_program_from_file(name);
-    let program_keypair = program_keypair.unwrap_or_else(|| {
-        let program_keypair = Keypair::new();
-        let instruction = system_instruction::create_account(
-            &payer_keypair.pubkey(),
-            &program_keypair.pubkey(),
-            1.max(
-                bank_client
-                    .get_minimum_balance_for_rent_exemption(program.len())
-                    .unwrap(),
-            ),
-            program.len() as u64,
-            loader_id,
-        );
-        let message = Message::new(&[instruction], Some(&payer_keypair.pubkey()));
-        bank_client
-            .send_and_confirm_message(&[payer_keypair, &program_keypair], message)
-            .unwrap();
-        program_keypair
-    });
-    let chunk_size = CHUNK_SIZE;
-    let mut offset = 0;
-    for chunk in program.chunks(chunk_size) {
-        let instruction =
-            loader_instruction::write(&program_keypair.pubkey(), loader_id, offset, chunk.to_vec());
-        let message = Message::new(&[instruction], Some(&payer_keypair.pubkey()));
-        bank_client
-            .send_and_confirm_message(&[payer_keypair, &program_keypair], message)
-            .unwrap();
-        offset += chunk_size as u32;
-    }
-    let instruction = loader_instruction::finalize(&program_keypair.pubkey(), loader_id);
-    (program_keypair, instruction)
-}
-
-pub fn load_program<T: Client>(
```

Comment:
Was this function unused?

Reply:
Yes they were for the loaders-v1 and v2, which we deprecated.

Reply:
So these are pure deletion of the code. The new functions are not related to these.
(trying to avoid comparing two independent functions).

Reply:
Yes, implementation / code wise they are unrelated. Conceptually they are related because these are their replacements but for the newest loader.